### ISSUE TITLE
The phase of Channel Permutation operation from ASP

### ISSUE THREAD

**Issue by erwangccc** [NVIDIA/apex#1666]
Hi, @jpool-nv @ChongyuNVIDIA 

Thanks for your great work first. 
And i'm using this feature in order to maintain accuracy. The channel permutation should be process at step 2 with Pruning under my understanding, The dense training, pruning, then sparsity training, like following pic.

![image](https://github.com/NVIDIA/apex/assets/47174977/818b32df-11ab-4521-8928-1f0c09f0019c)

However the _Network accuracy_ at chapter 5 from [paper](https://proceedings.neurips.cc/paper_files/paper/2021/file/6e8404c3b93a9527c8db241a1846599a-Paper.pdf), it is said to train densely first, pruning, then sparsity training, and then perform channel permute. 

> First, we train each network densely from scratch, using
hyperparameters from public repositories (see Appendix A). Then, we prune the dense models with
2:4 sparsity and repeat the training process again, resetting hyperparameters and optimizer state [22].
Finally, we search for and apply permutations (using the technique from Section 4.4) to each layer of
the dense models, then prune and repeat the training process to fine-tune the remaining weights.

So i wonder that which approach should i follow? Thanks in advance.

**Comment by jpool-nv** [comment]
Hi @erwangccc , your understanding is correct - the picture you've included illustrates the proper order of processes: dense training, permutations+pruning, then fine-tuning.  In the paper, we described the full experiment, in which we compared pruning+fine-tuning a model both with and without channel permutations.  Here's how I'd re-write the paragraph to clarify:

> First, we train each network densely from scratch, using hyperparameters from public repositories (see Appendix A). Then, **to determine the baseline 2:4 sparse accuracy without permutations,** we prune the dense models with 2:4 sparsity and repeat the training process again, resetting hyperparameters and optimizer state [22].  Finally, **to show the benefit of using channel permutations before the pruning step,** we search for and apply permutations (using the technique from Section 4.4) to each layer of the dense models, then prune and repeat the training process to fine-tune the remaining weights.

I hope it's clear now that the paper was describing two different experimental procedures in that paragraph, but let me know if I can explain further.

**Comment by erwangccc** [comment]
@jpool-nv Thanks for your reply. 
I understood the meaning after reading the rewrite paragraph.

**Comment by erwangccc** [comment]
Hi, @jpool-nv , continue to ask on this issue.

If i use 4:8 structured sparse, and i still want to use channel permutation to maintain accuracy, should i change the code to adopt the 4:8 structured sparse, and what code should i modify？

Thx～

**Comment by erwangccc** [comment]
@jpool-nv Thanks for reply.

After reading the code, I really think that it is a relatively large workload. I'm going to think about it.
